src/registry.rs

```diff
@@ -228,41 +258,58 @@ impl Registry {
             .expect("sub_registries not to be empty.")
     }
 
-    /// [`Iterator`] over all metrics registered with the [`Registry`].
-    pub fn iter(&self) -> RegistryIterator {
```

Comment:
Renaming `RegistryIterator` and making this `pub(crate)` is the only breaking change as far as I can tell. Might be worth undoing.